UnitsNet.Tests/QuantityIFormattableTests.cs

```diff
@@ -65,6 +65,36 @@ public void SFormatEqualsSignificantDigits(string sFormatString, string expected
             Assert.Equal(expected, length.ToString(sFormatString, NumberFormatInfo.InvariantInfo));
         }
 
+        /// <summary>
+        /// This verifies that the culture is correctly considered when formatting objects with an explicit culture.
+        /// </summary>
+        [Fact]
+        public void FormattingUsesSuppliedLocale()
+        {
+            Temperature t = Temperature.FromDegreesCelsius(2012.1234);
+            CultureInfo c = new CultureInfo("de-CH", false);
+            string formatted = string.Format(c, "{0:g}", t);
+            // Let's be very explicit here
+            Assert.Equal("2" + c.NumberFormat.NumberGroupSeparator + "012" + c.NumberFormat.NumberDecimalSeparator + "12 °C", formatted);
+        }
+
+        /// <summary>
+        /// This verifies that the culture is correctly considered when using <see cref="FormattableString.ToString(IFormatProvider)"/>
+        /// </summary>
+        [Fact]
+        public void FormatStringWorksWithSuppliedLocale()
+        {
+            Temperature t = Temperature.FromDegreesCelsius(2012.1234);
+            CultureInfo c = new CultureInfo("de-CH", false);
+
+            FormattableString f = $"{t:g}";
+            Assert.Equal("2" + c.NumberFormat.NumberGroupSeparator + "012" + c.NumberFormat.NumberDecimalSeparator + "12 °C", f.ToString(c));
+
+            // This does not work. Looks like a compiler bug to me.
+            // string f2 = $"{t:g}".ToString(c);
+            // Assert.Equal("2'012.12 °C", f2.ToString(c)); // Actual value is formatted according to CurrentUiCulture.
```

Comment:
The commented code does not work, because these two statements are equivalent:
```c#
string f2 = $"{t:g}".ToString(c);
string f2 = t.ToString("g").ToString(c);
```

### LINQPad sample
when CurrentUICulture is Norwegian and thousand separator is comma `,`.
```c#
Temperature t = Temperature.FromDegreesCelsius(2012.1234);
CultureInfo c = new CultureInfo("de-CH", false);

FormattableString f = $"{t:g}";
string f2 = $"{t:g}".ToString(c);
string f3 = t.ToString("g").ToString(c);

var expected = "2" + c.NumberFormat.NumberGroupSeparator + "012" + c.NumberFormat.NumberDecimalSeparator + "12 °C";
(expected == f.ToString(c).Dump()).Dump();
(expected == f2.ToString(c).Dump()).Dump(); // Actual value is formatted according to CurrentUiCulture.
(expected == f3.ToString(c).Dump()).Dump(); // Actual value is formatted according to CurrentUiCulture.
```

Output:
```
2’012.12 °C
True
2,012.12 °C
False
2,012.12 °C
False
```

Reply:
Remove the commented code from this PR.

Reply:
As you've seen 
```
FormattableString f = $"{t:g}";
string result = f.ToString(c);
```
is not equivalent to
```
string result = $"{t:g}".ToString(c);
```
which is what confused me (as this would be the case for every other expression).

I'll remove it, since the problem has nothing to do with this lib, but is a (weird but documented) feature of the CLR.